core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -754,7 +743,6 @@ private[spark] class TaskSetManager(
     }
     removeRunningTask(tid)
     info.markFinished(state)
-    successfulTaskIdsSet -= tid
```

Comment:
Removing duration from `successfulTaskDurations` is quite time consuming(O(n)) now. We just use `successfulTaskDurations` to generate the median duration. I might hesitate to do the remove.